parentchain/block-importer/src/parentchain_block_importer.rs

```diff
@@ -0,0 +1,230 @@
+	/// * sends `confirm_call` xt's of the executed unshielding calls
+	/// * sends `confirm_blocks` xt's for every imported parentchain block
+	fn import_parentchain_blocks(
+		&self,
+		blocks_to_import: Vec<SignedBlockG<Self::ParentchainBlockType>>,
+	) -> Result<()>;
+}
+
+/// Parentchain block import implementation.
+pub struct ParentchainBlockImporter<
+	PB,
+	ValidatorAccessor,
+	OCallApi,
+	StfExecutor,
+	ExtrinsicsFactory,
+	IndirectCallsExecutor,
+> where
+	PB: BlockT<Hash = H256>,
+	NumberFor<PB>: BlockNumberOps,
+	ValidatorAccessor: ValidatorAccess<PB>,
+	OCallApi: EnclaveOnChainOCallApi + EnclaveAttestationOCallApi,
+	StfExecutor: StfUpdateState + StfExecuteTrustedCall + StfExecuteShieldFunds,
+	ExtrinsicsFactory: CreateExtrinsics,
+	IndirectCallsExecutor: ExecuteIndirectCalls,
+{
+	validator_accessor: Arc<ValidatorAccessor>,
+	ocall_api: Arc<OCallApi>,
+	stf_executor: Arc<StfExecutor>,
+	extrinsics_factory: Arc<ExtrinsicsFactory>,
+	indirect_calls_executor: Arc<IndirectCallsExecutor>,
+	_phantom: PhantomData<PB>,
+}
+
+impl<PB, ValidatorAccessor, OCallApi, StfExecutor, ExtrinsicsFactory, IndirectCallsExecutor>
+	ParentchainBlockImporter<
+		PB,
+		ValidatorAccessor,
+		OCallApi,
+		StfExecutor,
+		ExtrinsicsFactory,
+		IndirectCallsExecutor,
+	> where
+	PB: BlockT<Hash = H256, Header = ParentchainHeader>,
+	NumberFor<PB>: BlockNumberOps,
+	ValidatorAccessor: ValidatorAccess<PB>,
+	OCallApi: EnclaveOnChainOCallApi + EnclaveAttestationOCallApi,
+	StfExecutor: StfUpdateState + StfExecuteTrustedCall + StfExecuteShieldFunds,
+	ExtrinsicsFactory: CreateExtrinsics,
+	IndirectCallsExecutor: ExecuteIndirectCalls,
+{
+	pub fn new(
+		validator_accessor: Arc<ValidatorAccessor>,
+		ocall_api: Arc<OCallApi>,
+		stf_executor: Arc<StfExecutor>,
+		extrinsics_factory: Arc<ExtrinsicsFactory>,
+		indirect_calls_executor: Arc<IndirectCallsExecutor>,
+	) -> Self {
+		ParentchainBlockImporter {
+			validator_accessor,
+			ocall_api,
+			stf_executor,
+			extrinsics_factory,
+			indirect_calls_executor,
+			_phantom: Default::default(),
+		}
+	}
+}
+
+impl<PB, ValidatorAccessor, OCallApi, StfExecutor, ExtrinsicsFactory, IndirectCallsExecutor>
+	ImportParentchainBlocks
+	for ParentchainBlockImporter<
+		PB,
+		ValidatorAccessor,
+		OCallApi,
+		StfExecutor,
+		ExtrinsicsFactory,
+		IndirectCallsExecutor,
+	> where
+	PB: BlockT<Hash = H256, Header = ParentchainHeader>,
+	NumberFor<PB>: BlockNumberOps,
+	ValidatorAccessor: ValidatorAccess<PB>,
+	OCallApi: EnclaveOnChainOCallApi + EnclaveAttestationOCallApi,
+	StfExecutor: StfUpdateState + StfExecuteTrustedCall + StfExecuteShieldFunds,
+	ExtrinsicsFactory: CreateExtrinsics,
+	IndirectCallsExecutor: ExecuteIndirectCalls,
+{
+	type ParentchainBlockType = PB;
+
+	fn import_parentchain_blocks(
+		&self,
+		blocks_to_import: Vec<SignedBlockG<Self::ParentchainBlockType>>,
+	) -> Result<()> {
+		let mut calls = Vec::<OpaqueCall>::new();
```

Comment:
extracted from the `enclave-runtime`. Used to be called `sync_parentchain_...` - re-named it to `import_..` - to be consistent with sidechain naming. Does that make sense for you?